internal/pkg/api/handleAck.go

```diff
@@ -364,10 +365,18 @@ func (ack *AckT) updateAPIKey(ctx context.Context,
 	if apiKeyID != "" {
 		res, err := ack.bulk.APIKeyRead(ctx, apiKeyID, true)
 		if err != nil {
-			zlog.Error().
-				Err(err).
-				Str(LogAPIKeyID, apiKeyID).
-				Msg("Failed to read API Key roles")
+			if ack.isAPIKeyReadError(ctx, zlog, agentID, err) {
```

Comment:
a possible concern here is that, now for each agent, for each error, for each output in the policy
https://github.com/elastic/fleet-server/blob/c99ccd81baa26b75e55a3b71b7d65be4111d555c/internal/pkg/api/handleAck.go#L337
this could do an additional request to elasticsearch (waiting on bulk flushes)?
is this a correct understanding?

Reply:
yes that's a good point.